LandSandBoat/server
# Fix Ul'xzomit roaming and spawn behaviors

**_I affirm:_**

## What does this pull request do?

Fixes #2668 

Make baby Ul'xzomits spawn normally.

Ul'xzomit babies will now roam just like mother Ul'xzomits when the mother is dead. They will re-attach when the mother respawns, and will re-attach if the mother is alive when they spawn.

Reduce Ul'xzomit respawn time to be in-line with retail.

Fix a bug where Ul'xzomit babies could be immune to physical damage (seemed to be a spawn race condition I couldn't quite figure out).

Changed some defaults of `xi.follow.follow()` params, and added option to disable warping.

## Steps to test these changes

1. Go to Al'taieu.
2. Kill a baby Ul'xzomit and respawn it. Observe it automatically re-attaches.
3. Kill a mother Ul'xzomit. Observe the baby Ul'xzomits start roaming by themselves.
4. Kill a baby Ul'xzomit and respawn it. Observe it starts roaming by itself.
5. Respawn the mother Ul'xzomit. Observe the babies start following it again.
6. Fight a mother Ul'xzomit. Kill a baby Ul'xzomit and respawn it. Observe the baby only links when it gets close.
7. Kill an Ul'xzomit and let it respawn normally. Observe its respawn time is now 5.5 minutes.

## Other thoughts

~~I believe baby Ul'xzomits when following the mother have the `xi.pathFlag.RUN` flag set, despite my not setting it. My observations suggest their follow speed should be the normal roam speed. On retail, if they fall behind the mother they don't catch up unless the mother changes directions.~~

~~Baby Ul'xzomits might be linking *slightly* farther than they do in retail. It's harder to test since the babies move a little too fast.~~

~~The following should probably place a limit on its max pathfinding distance it calculates at one time. If the baby Ul'xzomits are far from the mother, the mother could have moved quite far by the time the baby gets there.~~

Ul'xzomits when roaming seem to occasionally despawn on their own (they do not on retail), perhaps by getting too far from their spawn. This is a general problem with LSB's roaming behavior, and not specific to the Ul'xzomit behavior implemented in this PR.

When a baby Ul'xzomit delatches, it sometimes moves in a straight line quite some distance to return to its "area" which isn't necessarily the same as the mother. This is a general problem in LSB's roaming behavior, and not specific to the Ul'xzomit behavior implemented in this PR.

Ul'aerns, Ul'hpemdes, and Ul'phuabos also seem to have 5.5 minute respawns on retail. ~~I haven't checked Om' or other mobs in Al'taieu, but~~ I suspect most of them have 5.5 minute respawns since Al'taieu is an outdoor zone (though you cannot mount). (See #3630)
